### PR TITLE
localAddress or proxy config is lost when redirecting

### PR DESCRIPTION
**Pull request by 0x4139**
the localAddress or proxy configuration is lost when redirecting because the request object is initialized with an empty configuration instead of inheriting the one that made the initial request

### CODE REVIEW

**Comment by simov** [comment]
@0x4139 it would be great if you can provide a test case as well.

**Comment by 0x4139** [comment]
it seems that i can't bind on any interface in travis.

**Comment by simov** [comment]
First of all this project uses 2 space indentation, can you check that in your editor? After that you can run `npm test` on your local workstation and fix all of lint errors, before pushing to GitHub.

**Comment by nylen** [comment]
```js
'http://google.com', //redirects to 'http://google.com'
```

:confused: Maybe this should be `redirects to https://google.com`?

File: tests/test-localAddress.js [hunk 1]
```diff
@@ -1,27 +1,49 @@
...
+        localInterfaces[ifname].forEach(function (iface) {
+            if ('IPv4' !== iface.family || iface.internal !== false) {
+                // skip over internal (i.e. 127.0.0.1) and non-ipv4 addresses
+                return;
+            }
+                localIPS.push(iface.address);
+        });
+    });
+    request.get({
+        uri: 'http://google.com', //redirects to 'http://google.com'
```
Comment (nylen) [inline tests/test-localAddress.js]: :confused: Maybe this should be `redirects to https://google.com`?

Comment (0x4139) [inline tests/test-localAddress.js]: fair enough, the problem is that travis doesn't have interfaces and the tests are failing :( , they are passing on my local machine but not on travis

Comment (simov) [inline tests/test-localAddress.js]: @0x4139 did you ever checked the travis-ci errors https://travis-ci.org/request/request/jobs/51369942 ?

Comment (nylen) [inline tests/test-localAddress.js]: or run `npm test` locally

**Comment by 0x4139** [comment]
@simov  sorry did not run npm test my bad.

**Comment by 0x4139** [comment]
Certificates are expired

**Comment by simov** [comment]
@0x4139 can you pull https://github.com/simov/request/commit/1f6fbd57b8e2be0071e9771846790dda137ebb9a here? I removed the `proxy` parameter as there is a check for it inside the init method, and also added one for the `localAddress` After that this is a :+1:

**Comment by simov** [comment]
Closing in favor of https://github.com/request/request/pull/1460

**Pull request closed without merging**